### PR TITLE
Introduce a quality gate experiment for "usual" logs tailing

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit introduces a quality gates experiments meant to demonstrate that logs Agent will not lose logs and will not consume too much memory when operating in a "usual" logs tailing situation. Logs are produced in two sources at 500 KiB/second and rotate 5 total times at 50 MiB. Intake latency is 75ms.

### Motivation

Protect the 'usual' logs tailing scenario from performance regression. 

### Additional Notes

Exact details may change once we haggle some with AML.

### CODE REVIEW

File: test/regression/cases/quality_gate_logs/experiment.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+optimization_goal: egress_throughput
```
Comment (scottopell) [inline test/regression/cases/quality_gate_logs/experiment.yaml:1]: would CPU be more useful here? The assertion on "no lost bytes" seems sufficient to ensure data is flowing through

Comment (blt) [reply test/regression/cases/quality_gate_logs/experiment.yaml:1]: Ah or memory, yes. I think the expectation here is we'll adequately keep up.

Comment (blt) [reply test/regression/cases/quality_gate_logs/experiment.yaml]: Adjusted to CPU in [9be4e06](https://github.com/DataDog/datadog-agent/pull/31326/commits/9be4e06f2aba2b92ce613accd70d78b1ddfef5db).

**Review by gh123man — APPROVED** [review]

**Review by goxberry — APPROVED** [review]
I think if this experiment's name starts with `quality_gate`, it may become mandatory in Agent CI -- we might want to check that before merge.

**Comment by goxberry** [comment]
> quality_gate_idle, bounds check memory_usage: 9/10 replicas passed. Failed 1 which is > 0. Gate FAILED.
quality_gate_logs, bounds check lost_bytes: 10/10 replicas passed. Gate passed.
quality_gate_logs, bounds check memory_usage: 10/10 replicas passed. Gate passed.
quality_gate_idle_all_features, bounds check memory_usage: 10/10 replicas passed. Gate passed.

^ Confirms that the new experiments will become mandatory.

**Review by scottopell — APPROVED** [review]

**Pull request merged**